torch_struct/distributions.py

```diff
@@ -204,26 +244,30 @@ def partition(self):
         "Compute the log-partition function."
         return self._struct(LogSemiring).sum(self.log_potentials, self.lengths)
 
-    def sample(self, sample_shape=torch.Size()):
+    def sample(self, sample_shape=torch.Size(), batch_size=10):
         r"""
         Compute structured samples from the distribution :math:`z \sim p(z)`.
 
         Parameters:
             sample_shape (int): number of samples
+            batch_size (int): number of samples to compute at a time
 
         Returns:
             samples (*sample_shape x batch_shape x event_shape*)
         """
-        assert len(sample_shape) == 1
-        nsamples = sample_shape[0]
+        if type(sample_shape) == int:
+            nsamples = sample_shape
+        else:
+            assert len(sample_shape) == 1
+            nsamples = sample_shape[0]
         samples = []
         for k in range(nsamples):
-            if k % 10 == 0:
+            if k % batch_size == 0:
```

Comment:
Oh yeah, sorry this is my fault. 10 is a global constant. Let's put it on MultiSampledSemiring.